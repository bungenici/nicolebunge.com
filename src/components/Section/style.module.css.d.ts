export const section: string;
export const sectionFull: string;
