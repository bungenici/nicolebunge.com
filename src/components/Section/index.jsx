import PropTypes from 'prop-types';
import React, { PureComponent } from 'react';
import styles from './style.module.css';

class Section extends PureComponent {
  static propTypes = {
    children: PropTypes.node,
  };

  static defaultProps = {
    children: null,
  };

  render() {
    const { children } = this.props;

    return (
      <section className={styles.section}>
        {children}
      </section>
    );
  }
}

export default Section;
