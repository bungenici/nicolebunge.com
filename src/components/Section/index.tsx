import classNames from 'classnames';
import React from 'react';
import styles from './style.module.css';

export interface SectionProps {
  className?: string;
  full?: boolean;
}

const Section: React.FC<SectionProps> = (props) => {
  const { children, className, full, ...otherProps } = props;

  return (
    <section
      className={classNames(className, styles.section, {
        [styles.sectionFull]: full,
      })}
      {...otherProps}
    >
      {children}
    </section>
  );
};

export default Section;
