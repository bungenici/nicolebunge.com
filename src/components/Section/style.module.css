.section {
  align-items: center;
  display: flex;
  min-height: 100vh;
  padding-bottom: 6em;
  padding-top: 6em;
}
