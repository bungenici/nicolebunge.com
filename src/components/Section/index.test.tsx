/* eslint-disable import/no-extraneous-dependencies */

import React from 'react';
import renderer from 'react-test-renderer';
import Section from '.';

describe('Section component', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Section />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
