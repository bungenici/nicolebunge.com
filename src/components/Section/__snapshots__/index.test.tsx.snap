// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section component renders correctly 1`] = `
<DocumentFragment>
  <section
    class="section"
  />
</DocumentFragment>
`;
